pkg/console/subresource/deployment/deployment_test.go

```diff
@@ -1,170 +1,166 @@
-					UID:                        "",
-					ResourceVersion:            "",
-					Generation:                 0,
-					CreationTimestamp:          metav1.Time{},
-					DeletionTimestamp:          nil,
-					DeletionGracePeriodSeconds: nil,
-					Labels:          labels,
-					Annotations:     nil,
-					OwnerReferences: nil,
-					Initializers:    nil,
-					Finalizers:      nil,
-					ClusterName:     "",
-				},
-				Spec: appsv1.DeploymentSpec{
-					Replicas: &replicaCount,
-					Selector: &metav1.LabelSelector{
-						MatchLabels: labels,
-					},
-					Template: corev1.PodTemplateSpec{ObjectMeta: metav1.ObjectMeta{
-						Name:   api.OpenShiftConsoleShortName,
-						Labels: labels,
-						Annotations: map[string]string{
-							configMapResourceVersionAnnotation:          "",
-							secretResourceVersionAnnotation:             "",
-							serviceCAConfigMapResourceVersionAnnotation: "",
-						},
-					},
-						Spec: corev1.PodSpec{
-							// we want to deploy on master nodes
-							NodeSelector: map[string]string{
-								// empty string is correct
-								"node-role.kubernetes.io/master": "",
-							},
-							Affinity: &corev1.Affinity{
-								// spread out across master nodes rather than congregate on one
-								PodAntiAffinity: &corev1.PodAntiAffinity{
-									PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
-										Weight: 100,
-										PodAffinityTerm: corev1.PodAffinityTerm{
-											LabelSelector: &metav1.LabelSelector{
-												MatchLabels: util.SharedLabels(),
-											},
-											TopologyKey: "kubernetes.io/hostname",
-										},
-									}},
-								},
-							},
-							// toleration is a taint override. we can and should be scheduled on a master node.
-							Tolerations: []corev1.Toleration{
-								{
-									Key:      "node-role.kubernetes.io/master",
-									Operator: corev1.TolerationOpExists,
-									Effect:   corev1.TaintEffectNoSchedule,
-								},
-							},
-							RestartPolicy:                 corev1.RestartPolicyAlways,
-							SchedulerName:                 corev1.DefaultSchedulerName,
-							TerminationGracePeriodSeconds: &gracePeriod,
-							SecurityContext:               &corev1.PodSecurityContext{},
-							Containers: []corev1.Container{
-								consoleContainer(nil),
-							},
-							Volumes: consoleVolumes(volumeConfigList),
-						},
-					},
-					Strategy:                appsv1.DeploymentStrategy{},
-					MinReadySeconds:         0,
-					RevisionHistoryLimit:    nil,
-					Paused:                  false,
-					ProgressDeadlineSeconds: nil,
-				},
-				Status: appsv1.DeploymentStatus{},
-			},
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if got := DefaultDeployment(tt.args.cr, tt.args.cm, tt.args.cm, tt.args.sec); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("\nDefaultDeployment() = %v\n, want %v", got, tt.want)
-			}
-		})
-	}
-}
+//func TestDefaultDeployment(t *testing.T) {
```

Comment:
Output on this test was pretty hard to deal with, just the `reflect` / deep comparsion:

```
deployment_test.go:163:
    		DefaultDeployment() = &Deployment{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:openshift-console,GenerateName:,Namespace:openshift-console,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: openshift-console,component: ui,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:DeploymentSpec{Replicas:*2,Selector:&k8s_io_apimachinery_pkg_apis_meta_v1.LabelSelector{MatchLabels:map[string]string{app: openshift-console,component: ui,},MatchExpressions:[],},Template:k8s_io_api_core_v1.PodTemplateSpec{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:console,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: openshift-console,component: ui,},Annotations:map[string]string{console.openshift.io/console-config-version: ,console.openshift.io/oauth-secret-version: ,console.openshift.io/service-ca-config-version: ,},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:PodSpec{Volumes:[{console-serving-cert {nil nil nil nil nil SecretVolumeSource{SecretName:console-serving-cert,Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}} {console-oauth-config {nil nil nil nil nil &SecretVolumeSource{SecretName:console-oauth-config,Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}} {console-config {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:console-config,},Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil}} {service-ca {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:service-ca,},Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil}}],Containers:[{console  [/opt/bridge/bin/bridge --public-dir=/opt/bridge/static --config=/var/console-config/console-config.yaml --service-ca-file=/var/service-ca/service-ca.crt] []  [{https 0 443 TCP }] [] [] {map[] map[cpu:{{10 -3} {<nil>} 10m DecimalSI} memory:{{104857600 0} {<nil>} 100Mi BinarySI}]} [{console-serving-cert true /var/serving-cert  <nil>} {console-oauth-config true /var/oauth-config  <nil>} {console-config true /var/console-config  <nil>} {service-ca true /var/service-ca  <nil>}] [] Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/health,Port:8443,Host:,Scheme:HTTPS,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:30,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} &Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/health,Port:8443,Host:,Scheme:HTTPS,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:0,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} nil /dev/termination-log File IfNotPresent nil false false false}],RestartPolicy:Always,TerminationGracePeriodSeconds:*30,ActiveDeadlineSeconds:nil,DNSPolicy:,NodeSelector:map[string]string{node-role.kubernetes.io/master: ,},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:&PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[],},ImagePullSecrets:[],Hostname:,Subdomain:,Affinity:&Affinity{NodeAffinity:nil,PodAffinity:nil,PodAntiAffinity:&PodAntiAffinity{RequiredDuringSchedulingIgnoredDuringExecution:[],PreferredDuringSchedulingIgnoredDuringExecution:[{100 {LabelSelector{MatchLabels:map[string]string{app: openshift-console,},MatchExpressions:[],} [] kubernetes.io/hostname}}],},},SchedulerName:default-scheduler,InitContainers:[],AutomountServiceAccountToken:nil,Tolerations:[{node-role.kubernetes.io/master Exists  NoSchedule <nil>}],HostAliases:[],PriorityClassName:,Priority:nil,DNSConfig:nil,ShareProcessNamespace:nil,ReadinessGates:[],},},Strategy:DeploymentStrategy{Type:,RollingUpdate:nil,},MinReadySeconds:0,RevisionHistoryLimit:nil,Paused:false,ProgressDeadlineSeconds:nil,},Status:DeploymentStatus{ObservedGeneration:0,Replicas:0,UpdatedReplicas:0,AvailableReplicas:0,UnavailableReplicas:0,Conditions:[],ReadyReplicas:0,CollisionCount:nil,},}
    		, want &Deployment{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:openshift-console,GenerateName:,Namespace:openshift-console,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: openshift-console,component: ui,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:DeploymentSpec{Replicas:nil,Selector:&k8s_io_apimachinery_pkg_apis_meta_v1.LabelSelector{MatchLabels:map[string]string{app: openshift-console,component: ui,},MatchExpressions:[],},Template:k8s_io_api_core_v1.PodTemplateSpec{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:console,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: openshift-console,component: ui,},Annotations:map[string]string{console.openshift.io/console-config-version: ,console.openshift.io/oauth-secret-version: ,console.openshift.io/service-ca-config-version: ,},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:PodSpec{Volumes:[{console-serving-cert {nil nil nil nil nil SecretVolumeSource{SecretName:console-serving-cert,Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}} {console-oauth-config {nil nil nil nil nil &SecretVolumeSource{SecretName:console-oauth-config,Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}} {console-config {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:console-config,},Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil}} {service-ca {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:service-ca,},Items:[],DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil}}],Containers:[{console  [/opt/bridge/bin/bridge --public-dir=/opt/bridge/static --config=/var/console-config/console-config.yaml --service-ca-file=/var/service-ca/service-ca.crt] []  [{https 0 443 TCP }] [] [] {map[] map[memory:{{104857600 0} {<nil>} 100Mi BinarySI} cpu:{{10 -3} {<nil>} 10m DecimalSI}]} [{console-serving-cert true /var/serving-cert  <nil>} {console-oauth-config true /var/oauth-config  <nil>} {console-config true /var/console-config  <nil>} {service-ca true /var/service-ca  <nil>}] [] Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/health,Port:8443,Host:,Scheme:HTTPS,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:30,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} &Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/health,Port:8443,Host:,Scheme:HTTPS,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:0,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} nil /dev/termination-log File IfNotPresent nil false false false}],RestartPolicy:Always,TerminationGracePeriodSeconds:*30,ActiveDeadlineSeconds:nil,DNSPolicy:,NodeSelector:map[string]string{node-role.kubernetes.io/master: ,},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:&PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[],},ImagePullSecrets:[],Hostname:,Subdomain:,Affinity:&Affinity{NodeAffinity:nil,PodAffinity:nil,PodAntiAffinity:&PodAntiAffinity{RequiredDuringSchedulingIgnoredDuringExecution:[],PreferredDuringSchedulingIgnoredDuringExecution:[{100 {LabelSelector{MatchLabels:map[string]string{app: openshift-console,},MatchExpressions:[],} [] kubernetes.io/hostname}}],},},SchedulerName:default-scheduler,InitContainers:[],AutomountServiceAccountToken:nil,Tolerations:[{node-role.kubernetes.io/master Exists  NoSchedule <nil>}],HostAliases:[],PriorityClassName:,Priority:nil,DNSConfig:nil,ShareProcessNamespace:nil,ReadinessGates:[],},},Strategy:DeploymentStrategy{Type:,RollingUpdate:nil,},MinReadySeconds:0,RevisionHistoryLimit:nil,Paused:false,ProgressDeadlineSeconds:nil,},Status:DeploymentStatus{ObservedGeneration:0,Replicas:0,UpdatedReplicas:0,AvailableReplicas:0,UnavailableReplicas:0,Conditions:[],ReadyReplicas:0,CollisionCount:nil,},}
time="2019-01-30T00:10:55-05:00" level=info msg="console.openshift.io/console-config-version changed? false (3.11 vs 3.11)"
time="2019-01-30T00:10:55-05:00" level=info msg="console.openshift.io/console-config-version changed? true (4.0 vs 3.11)"
time="2019-01-30T00:10:55-05:00" level=info msg="console.openshift.io/oauth-secret-version changed? false (3.11 vs 3.11)"
time="2019-01-30T00:10:55-05:00" level=info msg="console.openshift.io/oauth-secret-version changed? true (4.0 vs 3.11)"
FAIL
FAIL	github.com/openshift/console-operator/pkg/console/subresource/deployment	0.041s
```
I'm ok with commenting it out for this PR & fixing in a follow (since these unit tests just merged) as it migth take a little bit of reworking the output.

Reply:
Testing the tests :)

Reply:
@zherman0 per that quick slack chat